File: android/feature/feedback/src/main/java/com/escala/ministerial/feature/feedback/domain/model/Feedback.kt
package com.escala.ministerial.feature.feedback.domain.model

import java.time.LocalDateTime

data class Feedback(
    val id: Long,
    val ministroId: Long,
    val ministroNome: String,
    val eventoId: Long,
    val eventoNome: String,
    val nota: Int,
    val comentario: String?,
    val dataEnvio: LocalDateTime?,
    val status: StatusFeedback,
    val resposta: String?,
)

enum class StatusFeedback(val label: String) {
    PENDENTE("Pendente"),
    RESPONDIDO("Respondido"),
    ARQUIVADO("Arquivado"),
}

File: ios/EscalaMinisterial/Models/Feedback.swift
import Foundation

struct Feedback: Codable, Identifiable {
    var id: Int?
    var ministroId: Int
    var ministroNome: String?
    var eventoId: Int
    var eventoNome: String?
    var nota: Int
    var comentario: String?
    var dataEnvio: String?
    var status: String
    var resposta: String?

    enum StatusFeedback: String, CaseIterable {
        case pendente   = "PENDENTE"
        case respondido = "RESPONDIDO"
        case arquivado  = "ARQUIVADO"

        var label: String {
            switch self {
            case .pendente:   return "Pendente"
            case .respondido: return "Respondido"
            case .arquivado:  return "Arquivado"
            }
        }
    }

    var statusEnum: StatusFeedback? { StatusFeedback(rawValue: status) }

    var dataEnvioFormatada: String? {
        guard let d = dataEnvio else { return nil }
        // Try datetime first (ISO 8601)
        if let date = DateUtils.parseDateTime(d) {
            return DateUtils.formatDateTime(date)
        }
        return d
    }
}
